allauth/account/stages.py

```diff
@@ -1,11 +1,13 @@
+from typing import Optional
+
 from allauth.account.adapter import get_adapter
 from allauth.account.app_settings import EmailVerificationMethod
 from allauth.account.utils import resume_login, stash_login, unstash_login
 from allauth.utils import import_callable
 
 
 class LoginStage:
-    key = None
+    key: Optional[str] = None  # Set in subclasses
```

Comment:
Given a `LoginStage` instance, its `key` property can/should never be `None`. So typing wise, it should be `: str`, not `Optional`.